pkg/network/protocols/events/configuration.go

```diff
@@ -8,45 +8,194 @@
 package events
 
 import (
+	"math"
 	"os"
 	"sync"
 
 	manager "github.com/DataDog/ebpf-manager"
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/asm"
+	"github.com/cilium/ebpf/features"
 
 	ddebpf "github.com/DataDog/datadog-agent/pkg/ebpf"
+	"github.com/DataDog/datadog-agent/pkg/network/usm/utils"
 	"github.com/DataDog/datadog-agent/pkg/util/kernel"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
-// handlerByProtocol holds a temporary reference to a `ddebpf.PerfHandler`
-// instance for a given protocol. This is done to simplify the usage of this
-// package a little bit, so a call to `events.Configure` can be later linked
-// to a call to `events.NewConsumer` without the need to explicitly propagate
-// any values. The map is guarded by `handlerMux`.
-var handlerByProtocol map[string]*ddebpf.PerfHandler
-var handlerMux sync.Mutex
+// defaultPerfBufferSize controls the amount of memory in bytes used *per CPU*
+// allocated for buffering perf event data
+var defaultPerfEventBufferSize = 16 * os.Getpagesize()
+
+// defaultPerfHandlerSize controls the size of the go channel that buffers perf
+// events (*ddebpf.PerfHandler). All perf events handled by this library have
+// fixed size (sizeof(batch_data_t)) which is ~4KB, so by choosing a value of
+// 100 we'll be buffering up to ~400KB of data in *Go* heap memory.
+const defaultPerfHandlerSize = 100
```

Comment:
Please write in the PR description if the size of the per-buffers changes (for better or for worse)